packages/trader/src/App/Components/Elements/market-countdown-timer.jsx

```diff
@@ -0,0 +1,212 @@
+        const whenMarketOpens = async (days_offset, target_symbol) => {
+            // days_offset is 0 for today, 1 for tomorrow, etc.
+            if (days_offset > days_to_check_before_exit) return {};
+            let remaining_time_to_open;
+            const target_date = moment(new Date()).add(days_offset, 'days');
+            const api_res = await getTradingTimes(target_date.format('YYYY-MM-DD'));
+            if (!api_res.api_initial_load_error) {
+                const { times } = getSymbol(target_symbol, api_res.trading_times);
+                const { open, close } = times;
+                const is_closed_all_day = open?.length === 1 && open[0] === '--' && close[0] === '--';
+                if (is_closed_all_day) {
+                    // check tomorrow trading times
+                    return whenMarketOpens(days_offset + 1, target_symbol);
+                }
+                const date_str = target_date.toISOString().substring(0, 11);
+                const getUTCDate = hour => new Date(`${date_str}${hour}Z`);
+                for (let i = 0; i < open?.length; i++) {
+                    const diff = +getUTCDate(open[i]) - +new Date();
+                    if (diff > 0) {
+                        remaining_time_to_open = +getUTCDate(open[i]);
+                        if (isMounted() && target_symbol === symbol) {
+                            return setWhenMarketOpens({ days_offset, opening_time: open[i], remaining_time_to_open });
+                        }
+                    }
+                }
+                whenMarketOpens(days_offset + 1, target_symbol);
+            }
+        };
+
+        whenMarketOpens(0, symbol);
+    }, [isMounted, symbol]);
+
+    React.useEffect(() => {
+        let timer;
+        if (when_market_opens?.remaining_time_to_open) {
+            timer = setTimeout(async () => {
+                setTimeLeft(calculateTimeLeft(when_market_opens?.remaining_time_to_open));
+                if (when_market_opens.remaining_time_to_open <= 1 && typeof setActiveSymbols === 'function')
+                    await setActiveSymbols();
+            }, 1000);
+        }
+        return () => {
+            if (timer) {
+                clearTimeout(timer);
+            }
+        };
+    }, [time_left, when_market_opens]);
+
+    let timer_components = '';
+
+    Object.keys(time_left).forEach(interval => {
+        if (interval === 'days') {
+            if (time_left.days) {
+                timer_components += `${time_left.days} ${localize(time_left.days > 1 ? 'days' : 'day')}`;
+            }
+        } else if (time_left[interval] !== 0) {
+            const value = time_left[interval];
+            timer_components +=
+                interval === 'hours'
+                    ? `${time_left.days ? ', ' : ''}${value} ${localize(time_left.hours > 1 ? 'hrs' : 'hr')}`
+                    : ` ${localize('and')} ${value} ${localize(time_left.minutes > 1 ? 'mins' : 'min')}`;
```

Comment:
Same here, please `localize` using individual calls to `localize` or by using placeholders.

Reply:
See above 🙏🏻

Reply:
I'm commenting on the usage of your `localize`: 

```JS
localize(time_left.minutes > 1 ? 'mins' : 'min'
```

Above won't work. It can only contain strings, you can't concatenate inside of it.